Cabal/Distribution/Simple/Program/Builtin.hs

```diff
@@ -309,6 +311,16 @@ hscolourProgram = (simpleProgram "hscolour") {
         _         -> ""
   }
 
+doctestProgram :: Program
+doctestProgram = (simpleProgram "doctest") {
```

Comment:
On second thought, that might actually not be sufficient, as cabal-install could be compiled by a different ghc than the one it is used with, unless I'm mistaken.